app/ide-desktop/lib/dashboard/src/components/AriaComponents/Text/useVisualTooltip.tsx

```diff
@@ -23,18 +24,40 @@ export interface VisualTooltipProps {
     aria.AriaPositionProps,
     'containerPadding' | 'offset' | 'placement'
   >
+  /**
+   * Determines when the tooltip should be displayed.
+   * - 'always': Tooltip is always displayed when the target element is hovered over.
+   * - 'when-overflowing': Tooltip is displayed only when the target element is overflowing.
+   * - A function that returns a boolean. The function is called with the target element as an argument.
+   */
+  readonly displayStrategy?: DisplayStrategy | ((target: HTMLElement) => boolean)
```

Comment:
nit: would prefer the function member to be `displayStrategyFn` which would make the API resemble static languages more

nit 2: consider calling it `display` or `shouldDisplay` (`display="always"`, `shouldDisplay="whenOverflowing"`)
nit 3: consider `whenTruncated` vs `whenOverflowing`